core/utils/cuckoo_map.h

```diff
@@ -180,7 +180,7 @@ class CuckooMap {
   Entry* Find(const K& key, const H& hasher = H(), const E& eq = E()) {
     // Blame Effective C++ for this
     return const_cast<Entry*>(
-        static_cast<const typeof(*this)&>(*this).Find(key, hasher, eq));
+        static_cast<const CuckooMap&>(*this).Find(key, hasher, eq));
```

Comment:
Everything I found says that `typeof` works the way changlan says. You'd have to provide a template that derived classes could instantiate to work around that, I think.